File: tests/syntax/supported/scala/contract.scala
package demo.core

import scala.collection.mutable.ListBuffer

@deprecated("legacy", "1.0")
trait Runner:
  def run[T](value: T): Int

enum State:
  case Ready, Done

class Worker(val name: String) extends Runner:
  private var count = 0

  def run[T](value: T): Int =
    val local = value
    println(s"Value $local")
    if local != null then 1 else 0

object Main:
  def main(args: Array[String]): Unit =
    val worker = new Worker("worker")
    worker.run(1)

File: tests/syntax/supported/ocaml/contract.ml
module type Runner = sig
  val run : 'a -> int
end

module Demo = struct
  type state = Ready | Done
  type person = { name : string; age : int }

  let count = ref 0

  let run value =
    let local = value in
    Printf.printf "Value %d\n" local;
    match local with
    | 0 -> Ready
    | _ -> Done

  class worker initial = object
    val mutable current = initial
    method get = current
  end
end
